WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift

```diff
@@ -401,6 +413,27 @@ private extension SettingsViewController {
         present(installJetpackController, animated: true, completion: nil)
     }
 
+    func shippingZonesWasPressed() {
+        guard let siteID = ServiceLocator.stores.sessionManager.defaultStoreID else {
+            return DDLogError("⛔️ Cannot SiteID to present shipping zones")
+        }
+
+        let tracksProvider = ReactNativeAnalyticsProvider()
+        let viewController: WCReactNativeViewController = {
+
+            if ServiceLocator.stores.isAuthenticatedWithoutWPCom {
+                let url = ServiceLocator.stores.sessionManager.defaultSite?.url ?? ""
+                let password = ApplicationPasswordEncoder().encodedPassword() ?? ""
+                return WCReactNativeViewController(analyticsProvider: tracksProvider, siteUrl: url, appPassword: password)
+            } else {
+                let token = Keychain(service: WooConstants.keychainServiceName).currentAuthToken ?? ""
+                return WCReactNativeViewController(analyticsProvider: tracksProvider, blogID: "\(siteID)", apiToken: token)
```

Comment:
Just curious, what are the reasons `WCReactNativeViewController` is in Objective-C? 😮 Is it required for Android, or just what we already had from Gutenberg before?

If this RN code is specific to WooCommerce, it might make sense to use `siteID` instead of `blogID`.

Reply:
By default [Native Modules docs and tutorials](https://reactnative.dev/docs/native-modules-ios) mentions only obj-c. However, upon your question, I see that we can have switch modules + some extra objc bridges. 

Honestly, I haven't spent much time with that because I wanted to avoid as many conflicts as possible but I'll take a look at it soon and see how much of a configuration hassle it is. I don't really like to write obj-c 😅